diff --git a/node_modules/react-native-purchases-ui/android/src/main/java/com/revenuecat/purchases/react/ui/RNCustomerCenterModule.kt b/node_modules/react-native-purchases-ui/android/src/main/java/com/revenuecat/purchases/react/ui/RNCustomerCenterModule.kt
index b0821b9..882fcc4 100644
--- a/node_modules/react-native-purchases-ui/android/src/main/java/com/revenuecat/purchases/react/ui/RNCustomerCenterModule.kt
+++ b/node_modules/react-native-purchases-ui/android/src/main/java/com/revenuecat/purchases/react/ui/RNCustomerCenterModule.kt
@@ -65,7 +65,7 @@ internal class RNCustomerCenterModule(
     fun presentCustomerCenter(
         promise: Promise
     ) {
-        currentActivity?.let {
+        reactContext.currentActivity?.let {
             customerCenterPromise = promise
             presentCustomerCenterFromActivity(it)
         } ?: run {
diff --git a/node_modules/react-native-purchases-ui/android/src/main/java/com/revenuecat/purchases/react/ui/RNPaywallsModule.kt b/node_modules/react-native-purchases-ui/android/src/main/java/com/revenuecat/purchases/react/ui/RNPaywallsModule.kt
index abadc93..562ae51 100644
--- a/node_modules/react-native-purchases-ui/android/src/main/java/com/revenuecat/purchases/react/ui/RNPaywallsModule.kt
+++ b/node_modules/react-native-purchases-ui/android/src/main/java/com/revenuecat/purchases/react/ui/RNPaywallsModule.kt
@@ -22,7 +22,7 @@ internal class RNPaywallsModule(
 
     private val currentFragmentActivity: FragmentActivity?
         get() {
-            return when (val currentActivity = currentActivity) {
+            return when (val currentActivity = reactApplicationContext.currentActivity) {
                 is FragmentActivity -> currentActivity
                 else -> {
                     Log.e(NAME, "RevenueCat paywalls require applications to use a FragmentActivity")
